CA-26187, CA-24021: Add an extra TIMER_SLOP when calculating HPET
deadlines, to avoid a race where the guest could read the HPET counter
after the interrupt and see a value just below the comparator.

diff -r fac36280b13a xen/arch/x86/hvm/hpet.c
--- a/xen/arch/x86/hvm/hpet.c	Fri Jul 09 13:40:42 2010 +0100
+++ b/xen/arch/x86/hvm/hpet.c	Fri Jul 09 13:41:49 2010 +0100
@@ -280,7 +280,8 @@
      */
     oneshot = !timer_is_periodic(h, tn);
     create_periodic_time(vhpet_vcpu(h), &h->pt[tn],
-                         hpet_tick_to_ns(h, diff),
+                         hpet_tick_to_ns(h, diff)
+                         /* XXX add TIMER_SLOP 50us */ + 50000,
                          oneshot ? 0 : hpet_tick_to_ns(h, h->hpet.period[tn]),
                          irq, NULL, NULL);
 }
